br/pkg/lightning/backend/local/duplicate.go

```diff
@@ -70,14 +70,84 @@ type DuplicateManager struct {
 	keyAdapter        KeyAdapter
 }
 
+type pendingIndexHandles struct {
+	// all 4 slices should have exactly the same length.
+	dataConflictInfos []errormanager.DataConflictInfo
+	indexNames        []string
+	handles           []tidbkv.Handle
+	rawHandles        [][]byte
+}
```

Comment:
```golang
type pendingIndexHandle struct {
	dataConflictInfo errormanager.DataConflictInfo
	indexName        string
	handle           tidbkv.Handle
	rawHandle        []byte
}

type pendingIndexHandles []pendingIndexHandle
```
Is this better?

Reply:
oh this is by design because of line 524.